ldm/generate.py

```diff
@@ -561,17 +561,22 @@ def apply_postprocessor(
             ):
         # retrieve the seed from the image;
         seed   = None
-        image_metadata = None
         prompt = None
 
         args   = metadata_from_png(image_path)
-        seed   = args.seed
-        prompt = args.prompt
-        print(f'>> retrieved seed {seed} and prompt "{prompt}" from {image_path}')
+        if opt.seed is not None:
+            seed = opt.seed
```

Comment:
What if `opt.seed` is negative?

Reply:
Well, this gets fixed in the generator, which also looks for negative seeds, but that's confusing.
Replaced with more readable:
```
        args = metadata_from_png(image_path)
        seed = opt.seed or args.seed
        if seed is None:
            seed = random.randrange(0, np.iinfo(np.uint32).max)
        elif seed <= 0:
            seed = random.randrange(0, np.iinfo(np.uint32).max)
```